lib/Backend/amd64/LowererMDArch.cpp

```diff
@@ -1098,6 +1098,17 @@ LowererMDArch::LowerAsmJsCallI(IR::Instr * callInstr)
 IR::Instr *
 LowererMDArch::LowerWasmMemOp(IR::Instr * instr, IR::Opnd *addrOpnd)
 {
+    IR::IndirOpnd * indirOpnd = addrOpnd->AsIndirOpnd();
+    IR::RegOpnd * indexOpnd = indirOpnd->GetIndexOpnd();
+    if (indexOpnd && !indexOpnd->IsIntegral32())
```

Comment:
do we not want to say uint32?

Reply:
I think it's possible for Globopt to change the index's type to int32 either through copy-prop or other mechanism